### PR TITLE
Support pushing span/trace through to splunk HEC

### PR DESCRIPTION
**Pull request by mr-miles**
I wanted to export spans to our splunk setup from the gateway but the way the chart is currently set up made it difficult to get all the pieces to match up (particularly port declarations across deployment/service/pods) without manually patching each one of them.

So this PR adds tracesIndex/tracesEnabled alongside metricsIndex/metricsEnabled etc so the traces are exported as expected.

Chart is working well in our environment.

### CODE REVIEW

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ clusterName: ""
 # Enterprise.
 splunkPlatform:
   # Required for Splunk Enterprise/Cloud. URL to a Splunk instance to send data
-  # to. e.g. "http://X.X.X.X:8088/services/collector". Setting this parameter
+  # to. e.g. "http://X.X.X.X:8088/services/collector".  Setting this parameter
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/values.yaml:27]:
```suggestion
  # to. e.g. "http://X.X.X.X:8088/services/collector". Setting this parameter
```

File: helm-charts/splunk-otel-collector/values.yaml [hunk 2]
```diff
@@ -67,7 +69,9 @@ splunkPlatform:
   logsEnabled: true
   # If you enable metrics collection, make sure that `metricsIndex` is provided as well.
   metricsEnabled: false
-
+  # If you enable traces collection, make sure that `tracesIndex` is provided as well.
+  tracesEnabled: false
+
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/values.yaml:74]:
```suggestion

```

**Comment by atoulme** [comment]
Thank you for the contribution, we appreciate it.

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 3]
```diff
@@ -194,7 +197,13 @@ service:
...
         - resource/add_environment
         {{- end }}
-      exporters: [sapm]
+      exporters:
+        {{- if (eq (include "splunk-otel-collector.o11yTracesEnabled" .) "true") }}
+        - sapm
+        {{- end }}
+        {{- if (eq (include "splunk-otel-collector.platformTracesEnabled" .) "true") }}
+        - splunk_hec/platform_traces
+        {{- end }}
```
Comment (atoulme) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:206]:
```suggestion
        {{- end }}
```

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 4]
```diff
@@ -194,7 +197,13 @@ service:
...
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:206]: Similar changes need to be done in the agent config. The gateway is not enabled by default

**Comment by dmitryax** [comment]
Thanks, @mr-miles. There is no reason to not enable tracing for Splunk Platform destination

**Comment by mr-miles** [comment]
@dmitryax good spot!  I updated the agent config (we were running in
gateway mode hence not spotting it)

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl (hunk 4 above)
Comment (mr-miles) [reply helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:206]: good spot!  I updated the agent config (we switched to running in gateway mode hence not spotting it)

**Review by dmitryax — APPROVED** [review]
LGTM. Thanks, @mr-miles

**Pull request merged**